lib/matplotlib/tests/test_backends_interactive.py

```diff
@@ -503,3 +502,46 @@ def test_blitting_events(env):
     # blitting is not properly implemented
     ndraws = proc.stdout.count("DrawEvent")
     assert 0 < ndraws < 5
+
+
+# The source of this function gets extracted and run in another process, so it
+# must be fully self-contained.
+def _test_figure_leak():
+    import sys
+
+    import psutil
+    from matplotlib import pyplot as plt
+    # Second argument is pause length, but if zero we should skip pausing
+    t = float(sys.argv[1])
+
+    # Warmup cycle, this reasonably allocates a lot
+    p = psutil.Process()
+    fig = plt.figure()
+    if t:
+        plt.pause(t)
+    plt.close(fig)
+    mem = p.memory_full_info().uss
+
+    for _ in range(5):
+        fig = plt.figure()
+        if t:
+            plt.pause(t)
+        plt.close(fig)
```

Comment:
In a standalone script, it appears that if you add a `gc.collect` at the end of the loop, that will fix Tk with 0.0 time. Strangely, adding a _single_ `gc.collect` at the end of everything doesn't free up everything.

For some reason, this does not work in the test, but it does cut the counted growth in half, and is definitely something we should add to the test.

Reply:
I added this.